contracts/transfer/tests/transfer.rs

```diff
@@ -553,3 +562,286 @@ fn phoenix_deposit_and_withdraw() {
         "Alice should have no balance after it is withdrawn"
     );
 }
+
+#[test]
+fn phoenix_to_moonlight_swap() {
+    const SWAP_VALUE: u64 = dusk(1.0);
+
+    let rng = &mut StdRng::seed_from_u64(0xfeeb);
+
+    let phoenix_sk = SecretKey::random(rng);
+    let phoenix_vk = ViewKey::from(&phoenix_sk);
+    let phoenix_pk = PublicKey::from(&phoenix_sk);
+
+    let moonlight_sk = BlsSecretKey::random(rng);
+    let moonlight_pk = BlsPublicKey::from(&moonlight_sk);
+
+    let vm = &mut rusk_abi::new_ephemeral_vm()
+        .expect("Creating ephemeral VM should work");
+    let mut session = &mut instantiate(rng, vm, &phoenix_pk, &moonlight_pk);
+
+    let swapper_account = account(&mut session, &moonlight_pk)
+        .expect("Getting account should succeed");
+
+    assert_eq!(
+        swapper_account.balance, MOONLIGHT_GENESIS_VALUE,
+        "The swapper's account should have the genesis value"
+    );
+
+    let leaves = leaves_from_height(session, 0)
+        .expect("getting the notes should succeed");
+    let notes = filter_notes_owned_by(
+        phoenix_vk,
+        leaves.into_iter().map(|leaf| leaf.note),
+    );
+
+    assert_eq!(notes.len(), 1, "There should be one note at this height");
+
+    let convert = Withdraw::new(
+        rng,
+        &moonlight_sk,
+        TRANSFER_CONTRACT.to_bytes(),
+        SWAP_VALUE,
+        WithdrawReceiver::Moonlight(moonlight_pk),
+        WithdrawReplayToken::Phoenix(vec![notes[0].gen_nullifier(&phoenix_sk)]),
+    );
+
+    let contract_call = ContractCall {
+        contract: TRANSFER_CONTRACT.to_bytes(),
+        fn_name: String::from("convert"),
+        fn_args: rkyv::to_bytes::<_, 1024>(&convert)
+            .expect("should serialize conversion correctly")
+            .to_vec(),
+    };
+
+    let tx = create_phoenix_transaction(
+        session,
+        &phoenix_sk,
+        &phoenix_pk,
+        GAS_LIMIT,
+        LUX,
+        [0],
+        42,
```

Comment:
Shouldn't this be a `0` since we don't actually 'transfer'

Reply:
True, copy paste issue. Will fix